### DISCUSSION TITLE
3+2 Axis Posts

### DISCUSSION THREAD

**Discussion by plopes9000** [TimPaterson/Fusion360-Batch-Post#53] [Ideas]
Just occured to me that by tagging the operations name we could achieve 3+2 axis, not simultaneous but still worth while.

For example:
   op1 name "myfolder: A:0 B:0 Handwheel"
   op2 name "myfolder: A:0 B:180 Handwheel"
   op3 name "myfolder: A:90 B:0 Handwheel"
   etc

Would result in a post on folder "myfolder" of one file "Handweel" where in between operations the A and B axis are moved as specified. We would also need some safe retract options in the dialog for the A, B or C axis moves in between ops.

What do you think ?

Cheers,
Paulo

**Comment by TimPaterson** [comment]
Couldn't you get what you want by creating a Manual NC operation and using Pass-through?

**Comment by plopes9000** [comment]
Actually what I proposed above was incorrect.

The sequence would need to be that for each A/B/C change there must be a dedicated "setup" with the XYZ configured as desired - so for A/C=0/0, Z would point up on the part, but for A/C=180/0 then Z is reversed. Only this way will F360 generate correct toolpaths for the different part orientations.

Hence, the bellow would output only one file called "Setup 1" issuing G0 A0 B0 and G0 A180 B0 before each setup ops + the safe XYZ retracts.

Setup 1: A:0 B:0
  op1 
  op2 
  op3 
Setup 1: A:180 B:0
  op1 
  op2 
... etc

Fusion360-Batch-Post does not generate single files for multiple setups, at most it can create one directory for multiple setups.

See below the test using manual NC

![3+2 setup 1 F360](https://user-images.githubusercontent.com/4532031/231093853-12343b85-0403-40c9-afa9-68632ad54646.png)
![3+2 setup 1](https://user-images.githubusercontent.com/4532031/231093879-1b069ce9-02fe-41a5-8798-9d33c831b0d3.png)

**Comment by plopes9000** [comment]
oh by the way thanks a lot for the plugin - life saver :-)

**Comment by plopes9000** [comment]
See https://github.com/TimPaterson/Fusion360-Batch-Post/compare/master...plopes9000:Fusion360-Batch-Post:3+2-support

just adds a way to replace custom strings per setup based on the setup name, e.g. Setup name: 'G0 A0. C0.=G0 A180. C0.#Shade:XYZA:Test'

results in posting to a file in Shade/XYZA/01 Test.ngc with the text 'G0 A0. C0.' replaced by 'G0 A180. C0.'

It is not elegant ... and it does not merge the setups into one output post.

Not really finding an elegant way to do this better. Also have not thought through the merging of multiple setups into a single post.
Could be an option on the dialog saying something along the lines of "Merge setups with same folder and name into single post file".

**Comment by TimPaterson** [comment]
I have no idea what you're talking about, but I'm glad you've made it work for you.

**Reply by plopes9000** [reply]
If you've not used 4/5 axis it would be lengthy to explain the entire reasoning - which I could do if it was of interest.
In any case, thanks again for writing this tool.
